### PR TITLE
DST Fix for energy_storage_resources

### PR DESCRIPTION
**Pull request by kmax12**
The problematic intervals
 
image (91)

image (92)

and the missing interval

image

### CODE REVIEW

File: gridstatus/ercot.py [hunk 1]
```diff
@@ -323,6 +335,9 @@ def get_energy_storage_resources(self, date="latest", verbose=False):
         )
 
         df = df.sort_values("Time").reset_index(drop=True)
+        import pdb
```
Comment (WillKoehrsen) [inline gridstatus/ercot.py:338]: Remove before merging.

File: gridstatus/ercot.py [hunk 2]
```diff
@@ -303,6 +303,18 @@ def get_energy_storage_resources(self, date="latest", verbose=False):
...
 
+        # todo(kanter): fix this for future DST dates
+        if "2024-11-03 02:00:00-0600" in df["timestamp"].values:
+            # ERCOT publishes two intervals with 2am timestamp
+            # during CDT to CST transition
+            # but skips the repeated 1am timestamp
+            # let's manually fix this before further timestamp parsing
+            df.loc[
+                (df["timestamp"] == "2024-11-03 02:00:00-0600")
+                & (df["dstFlag"] == "N"),
```
Comment (WillKoehrsen) [inline gridstatus/ercot.py:314]: This assumes the `dstFlag == "Y"` for the repeated hour only?

Comment (kmax12) [reply gridstatus/ercot.py:314]: not 100% I follow the question.  but there two 2am intervals with -6 offset but a missing 1am interval with -6 offset, so i am manually shifting one based on `dstFlag`

Comment (WillKoehrsen) [reply gridstatus/ercot.py:314]: How do you know to use "Y" or "N" to identify the interval to shift?

Comment (WillKoehrsen) [reply gridstatus/ercot.py:314]: If `dstFlag == "Y"` for the repeated hour and ERCOT labels rows by hour end then this logic makes sense.

**Review by WillKoehrsen — APPROVED** [review]

File: gridstatus/ercot.py (hunk 2 above)
Comment (kmax12) [reply gridstatus/ercot.py:314]: yeah, it is dstFlag == "Y" when things repeat. 

this data is 5 minute though. every other interval gets labeled correctly. this is some weird error on ERCOT's side  i think

**Pull request merged**